### PR TITLE
feat(uplink): add PodDisruptionBudget and unittests

### PR DESCRIPTION
**Pull request by smerle33**
as per https://github.com/jenkins-infra/helpdesk/issues/3719#issuecomment-1734939980

### CODE REVIEW

File: charts/uplink/templates/_helpers.tpl [hunk 1]
```diff
@@ -31,13 +31,21 @@ Create chart name and version as used by the chart label.
 {{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
 
+{{/*
+Selector labels
+*/}}
+{{- define "uplink.selectorLabels" -}}
+app.kubernetes.io/name: {{ include "uplink.name" . }}
```
Comment (dduportal) [inline charts/uplink/templates/_helpers.tpl:38]: Don't forget the unit test (on the deployment) for this change ;)

File: charts/uplink/templates/pdb.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{{- if (gt (int .Values.replicaCount) 1) }}
+apiVersion: policy/v1
+kind: PodDisruptionBudget
+metadata:
+  name: {{ include "uplink.fullname" . }}
+  labels:
+    {{- include "uplink.labels" . | nindent 4 }}
+spec:
+  {{- with .Values.poddisruptionbudget.minAvailable }}
```
Comment (dduportal) [inline charts/uplink/templates/pdb.yaml:9]: `.Values.poddisruptionbudget` is undefined causing the error `Error: Failed to render chart: exit status 1: Error: template: uplink/templates/pdb.yaml:9:18: executing "uplink/templates/pdb.yaml" at <.Values.poddisruptionbudget.minAvailable>: nil pointer evaluating interface {}.minAvailable`.

You might want to look at  https://github.com/jenkins-infra/helm-charts/pull/822/files and add the corresponding tests (first) and fix

File: charts/uplink/templates/_helpers.tpl (hunk 1 above)
Comment (smerle33) [reply charts/uplink/templates/_helpers.tpl:38]: done with https://github.com/jenkins-infra/helm-charts/pull/821/commits/003a84b7e106012ca1cf83255e858d72d772101c

**Review by dduportal — CHANGES_REQUESTED** [review]
Almost there! There is still one issue in the PDB though (see my comment)

**Pull request merged**